internal/pkg/checks/pathRulesCheck.go

```diff
@@ -135,4 +292,77 @@ var pathRulesBarSubpathCheck = &Check{
 	},
 }
 
-// TODO: Implement more checks on edge cases like leading `/`, query params, and encoding
+var pathRulesPrefixConsecutiveSlashesCheck = &Check{
+	Name:        "path-rules-prefix-consecutive-slashes",
+	Description: "Ingress with prefix path rule with consecutive slashes are ignored",
+	Run: func(check *Check, config Config) (success bool, err error) {
+		req, res, err := captureRequest(fmt.Sprintf("http://%s/routes/with/consecutive//slashes///are-ignored", pathRulesHost), "path-rules")
+		if err != nil {
+			return
+		}
+
+		a := new(assertionSet)
+		// Assert the request received from the downstream service
+		a.equals(req.TestId, "path-rules-catchall", "expected the downstream service would be '%s' but was '%s'")
+		a.equals(req.Path, "/routes/with/consecutive//slashes///are-ignored", "expected the request path would be '%s' but was '%s'")
```

Comment:
Given:
```
Every Path p_i must be syntactically valid:
  Must begin with the '/' character (relative paths are not allowed by RFC-7230).
  Must not contain consecutive '/' characters (e.g. /foo///, //).
```
I wanted to capture the fact a path such as `/routes/with/consecutive//slashes///are-ignored` was actually accepted by kubeapi, yet the ingress implementation should not consider it as valid and send downstream traffic to the catch-all service instead of `void`.

Ref: https://github.com/kubernetes/enhancements/blob/master/keps/sig-network/20190125-ingress-api-group.md#path-matching-semantics

Reply:
Maybe we should leave a TODO here, or file a separate issue, since I think this requirement is specific to ingress V1. We probably need a way to make checks conditional on the version of the resource.